src/equations/bbm_bbm_1d.jl

```diff
@@ -228,25 +234,27 @@ function rhs!(du_ode, u_ode, t, mesh, equations::BBMBBMEquations1D, initial_cond
     # energy and mass conservative semidiscretization
     if solver.D1 isa DerivativeOperator ||
        solver.D1 isa UniformCoupledOperator
-        @timeit timer() "deta hyperbolic" deta[:]=-solver.D1 * (D * v + eta .* v)
-        @timeit timer() "dv hyperbolic" dv[:]=-solver.D1 *
+        @timeit timer() "deta hyperbolic" tmp1=-solver.D1 * (D * v + eta .* v)
+        @timeit timer() "dv hyperbolic" tmp2=-solver.D1 *
                                               (equations.gravity * eta + 0.5 * v .^ 2)
     elseif solver.D1 isa UpwindOperators
-        @timeit timer() "deta hyperbolic" deta[:]=-solver.D1.minus * (D * v + eta .* v)
-        @timeit timer() "dv hyperbolic" dv[:]=-solver.D1.plus *
+        @timeit timer() "deta hyperbolic" tmp1=-solver.D1.minus * (D * v + eta .* v)
+        @timeit timer() "dv hyperbolic" tmp2=-solver.D1.plus *
                                               (equations.gravity * eta + 0.5 * v .^ 2)
     else
         @error "unknown type of first-derivative operator: $(typeof(solver.D1))"
     end
 
     @timeit timer() "source terms" calc_sources!(dq, q, t, source_terms, equations, solver)
 
-    @timeit timer() "deta elliptic" deta[:]=invImD2n * deta
+    @timeit timer() "deta elliptic" ldiv!(invImD2n, tmp1)
```

Comment:
This is still the 2-argument version of `ldiv!` - did this line lead to the error you mentioned?

Reply:
No, it shouldn't. This is for the reflecting bc, but up to now I'm only testing the periodic bc.

Reply:
From the stacktrace above we can see that the error occurs in line 209, which uses the three-argument `ldiv!` version.